It built surprisingly fast so here are the const qualifiers needed:

```
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx: In function 'int main(int, char**)':
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx:71:37: error: no match for 'operator=' (operand types are 'std::unique_ptr<GlobalParams>' and 'GlobalParams*')
     globalParams = new GlobalParams();
                                     ^
In file included from /nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/memory:80,
                 from /nix/store/p30m4hiihrh0dxly42ivg7w4h5dlmnxc-poppler-glib-0.83.0-dev/include/poppler/Link.h:34,
                 from /build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx:39,
                 from /build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx:20:
/nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/bits/unique_ptr.h:287:7: note: candidate: 'std::unique_ptr<_Tp, _Dp>& std::unique_ptr<_Tp, _Dp>::operator=(std::unique_ptr<_Tp, _Dp>&&) [with _Tp = GlobalParams; _Dp = std::default_delete<GlobalParams>]'
       operator=(unique_ptr&& __u) noexcept
       ^~~~~~~~
/nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/bits/unique_ptr.h:287:7: note:   no known conversion for argument 1 from 'GlobalParams*' to 'std::unique_ptr<GlobalParams>&&'
/nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/bits/unique_ptr.h:307:2: note: candidate: 'template<class _Up, class _Ep> typename std::enable_if<std::__and_<std::__and_<std::is_convertible<typename std::unique_ptr<_Up, _Ep>::pointer, typename std::__uniq_ptr_impl<_Tp, _Dp>::pointer>, std::__not_<std::is_array<_Up> >, std::__or_<std::__and_<std::is_reference<_Dp>, std::is_same<_T2, _U2> >, std::__and_<std::__not_<std::is_reference<_Dp> >, std::is_convertible<_Ep, _Dp> > > >, std::is_assignable<_T2&, _U2&&> >::value, std::unique_ptr<_Tp, _Dp>&>::type std::unique_ptr<_Tp, _Dp>::operator=(std::unique_ptr<_Up, _Ep>&&) [with _Up = _Up; _Ep = _Ep; _Tp = GlobalParams; _Dp = std::default_delete<GlobalParams>]'
  operator=(unique_ptr<_Up, _Ep>&& __u) noexcept
  ^~~~~~~~
/nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/bits/unique_ptr.h:307:2: note:   template argument deduction/substitution failed:
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx:71:37: note:   mismatched types 'std::unique_ptr<_Tp, _Dp>' and 'GlobalParams*'
     globalParams = new GlobalParams();
                                     ^
In file included from /nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/memory:80,
                 from /nix/store/p30m4hiihrh0dxly42ivg7w4h5dlmnxc-poppler-glib-0.83.0-dev/include/poppler/Link.h:34,
                 from /build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx:39,
                 from /build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx:20:
/nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/bits/unique_ptr.h:316:7: note: candidate: 'std::unique_ptr<_Tp, _Dp>& std::unique_ptr<_Tp, _Dp>::operator=(std::nullptr_t) [with _Tp = GlobalParams; _Dp = std::default_delete<GlobalParams>; std::nullptr_t = std::nullptr_t]'
       operator=(nullptr_t) noexcept
       ^~~~~~~~
/nix/store/vgn1sxcj5ma9m5bhjps1q9jx3q2ab27c-gcc-8.3.0/include/c++/8.3.0/bits/unique_ptr.h:316:7: note:   no known conversion for argument 1 from 'GlobalParams*' to 'std::nullptr_t'
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx: In member function 'virtual void pdfi::PDFOutDev::stroke(GfxState*)':
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:799:30: error: invalid conversion from 'const GfxPath*' to 'GfxPath*' [-fpermissive]
     printPath( state->getPath() );
                ~~~~~~~~~~~~~~^~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:494:37: note:   initializing argument 1 of 'static void pdfi::PDFOutDev::printPath(GfxPath*)'
 void PDFOutDev::printPath( GfxPath* pPath )
                            ~~~~~~~~~^~~~~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx: In member function 'virtual void pdfi::PDFOutDev::fill(GfxState*)':
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:810:30: error: invalid conversion from 'const GfxPath*' to 'GfxPath*' [-fpermissive]
     printPath( state->getPath() );
                ~~~~~~~~~~~~~~^~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:494:37: note:   initializing argument 1 of 'static void pdfi::PDFOutDev::printPath(GfxPath*)'
 void PDFOutDev::printPath( GfxPath* pPath )
                            ~~~~~~~~~^~~~~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx: In member function 'virtual void pdfi::PDFOutDev::eoFill(GfxState*)':
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:821:30: error: invalid conversion from 'const GfxPath*' to 'GfxPath*' [-fpermissive]
     printPath( state->getPath() );
                ~~~~~~~~~~~~~~^~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:494:37: note:   initializing argument 1 of 'static void pdfi::PDFOutDev::printPath(GfxPath*)'
 void PDFOutDev::printPath( GfxPath* pPath )
                            ~~~~~~~~~^~~~~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx: In member function 'virtual void pdfi::PDFOutDev::clip(GfxState*)':
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:832:30: error: invalid conversion from 'const GfxPath*' to 'GfxPath*' [-fpermissive]
     printPath( state->getPath() );
                ~~~~~~~~~~~~~~^~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:494:37: note:   initializing argument 1 of 'static void pdfi::PDFOutDev::printPath(GfxPath*)'
 void PDFOutDev::printPath( GfxPath* pPath )
                            ~~~~~~~~~^~~~~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx: In member function 'virtual void pdfi::PDFOutDev::eoClip(GfxState*)':
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:843:30: error: invalid conversion from 'const GfxPath*' to 'GfxPath*' [-fpermissive]
     printPath( state->getPath() );
                ~~~~~~~~~~~~~~^~
/build/libreoffice-6.3.0.4/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:494:37: note:   initializing argument 1 of 'static void pdfi::PDFOutDev::printPath(GfxPath*)'
 void PDFOutDev::printPath( GfxPath* pPath )
                            ~~~~~~~~~^~~~~
make[1]: *** [/build/libreoffice-6.3.0.4/solenv/gbuild/LinkTarget.mk:293: /build/libreoffice-6.3.0.4/workdir/CxxObject/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [/build/libreoffice-6.3.0.4/solenv/gbuild/LinkTarget.mk:293: /build/libreoffice-6.3.0.4/workdir/CxxObject/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.o] Error 1

```

--- a/pkgs/applications/office/libreoffice/default.nix
+++ b/pkgs/applications/office/libreoffice/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, pam, python3, libxslt, perl, ArchiveZip, gettext
+{ stdenv, fetchurl, fetchpatch, pam, python3, libxslt, perl, ArchiveZip, gettext
 , IOCompress, zlib, libjpeg, expat, freetype, libwpd
 , libxml2, db, curl, fontconfig, libsndfile, neon
 , bison, flex, zip, unzip, gtk3, gtk2, libmspack, getopt, file, cairo, which
@@ -73,6 +73,12 @@ in stdenv.mkDerivation rec {
 
   patches = [
     ./xdg-open-brief.patch
+    # poppler-0.82 compatibility:
+    (fetchpatch {
+      url = "https://github.com/LibreOffice/core/commit/2eadd46a.patch";
+      sha256 = "1mpipdfxvixjziizbhfbpybpzlg1ijw7s0yqjpmq5d7pf3pvkm4n";
+    })
+    ./poppler-0.83.patch
   ];
 
   tarballPath = "external/tarballs";

--- a/pkgs/applications/office/libreoffice/poppler-0.83.patch
+++ b/pkgs/applications/office/libreoffice/poppler-0.83.patch
@@ -1,0 +1,48 @@
+diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+index 26048177e87d..da7736f607f9 100644
+--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
++++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+@@ -491,12 +491,12 @@ void PDFOutDev::writeFontFile( GfxFont* gfxFont ) const
+     gfree(pBuf);
+ }
+ 
+-void PDFOutDev::printPath( GfxPath* pPath )
++void PDFOutDev::printPath( const GfxPath* pPath )
+ {
+     int nSubPaths = pPath ? pPath->getNumSubpaths() : 0;
+     for( int i=0; i<nSubPaths; i++ )
+     {
+-        GfxSubpath* pSub  = pPath->getSubpath( i );
++        const GfxSubpath* pSub  = pPath->getSubpath( i );
+         const int nPoints = pSub->getNumPoints();
+ 
+         printf( " subpath %d", pSub->isClosed() );
+diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
+index 02f6b59f6f15..1c7451a78601 100644
+--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
++++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
+@@ -149,7 +149,7 @@ namespace pdfi
+ 
+         int  parseFont( long long nNewId, GfxFont* pFont, GfxState* state ) const;
+         void writeFontFile( GfxFont* gfxFont ) const;
+-        static void printPath( GfxPath* pPath );
++        static void printPath( const GfxPath* pPath );
+ 
+     public:
+         explicit PDFOutDev( PDFDoc* pDoc );
+diff --git a/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx
+index 42178b650cdd..b1a54bd09c5f 100644
+--- a/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx
++++ b/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx
+@@ -68,7 +68,11 @@ int main(int argc, char **argv)
+     }
+ 
+     // read config file
++#if POPPLER_CHECK_VERSION(0, 83, 0)
++    globalParams = std::make_unique<GlobalParams>();
++#else
+     globalParams = new GlobalParams();
++#endif
+     globalParams->setErrQuiet(true);
+ #if defined(_MSC_VER)
+     globalParams->setupBaseFonts(nullptr);

--- a/pkgs/applications/office/libreoffice/still.nix
+++ b/pkgs/applications/office/libreoffice/still.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, pam, python3, libxslt, perl, ArchiveZip, gettext
+{ stdenv, fetchurl, fetchpatch, pam, python3, libxslt, perl, ArchiveZip, gettext
 , IOCompress, zlib, libjpeg, expat, freetype, libwpd
 , libxml2, db, curl, fontconfig, libsndfile, neon
 , bison, flex, zip, unzip, gtk3, gtk2, libmspack, getopt, file, cairo, which
@@ -73,6 +73,11 @@ in stdenv.mkDerivation rec {
 
   patches = [
     ./xdg-open-brief.patch
+    (fetchpatch {
+      url = "https://github.com/LibreOffice/core/commit/2eadd46a.patch";
+      sha256 = "1mpipdfxvixjziizbhfbpybpzlg1ijw7s0yqjpmq5d7pf3pvkm4n";
+    })
+    ./poppler-0.83.patch
   ];
 
   tarballPath = "external/tarballs";

